scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2389,15 +2349,15 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   }
 
   def condExprInParens[T <: Token: TokenInfo]: Term = {
-    def isFollowedBy[T <: Token: TokenInfo] = {
+    def isFollowedByT = {
```

Comment:
I would leave implicit in signature here as compiler might (or might not) optimize this function if it doesn't depend on parent function.

Reply:
this change is because
1. `isFollowedBy` uses the same symbol for the generic type (`T`) as the outer `condExprInParens`
2. however, instead of renaming it `U`, observe that `T` is exactly the same when used; and `condExprInParens` has all the implicits already.

Reply:
i can also rename to `U` or move this method out. let me know what you prefer.

Reply:
yep I've seen that.
As I remember in case of  inner `def` with a full signature compiler moves this `def` into synthetic one in class body by itself. Otherwise if it references to smth in outer def it will be a lambda.

I don't think that it there is a significant difference in performance but there is a point of doing that event if types and implicits are same.

> let me know what you prefer.

any option

Reply:
done. by the way, the method it calls (`followedByToken`) is not used anywhere else; perhaps this newline check should simply have been part of that method instead.